src/seq_mv/csr_matvec_device.c

```diff
@@ -48,6 +48,8 @@ hypre_CSRMatrixMatvecDevice2( HYPRE_Int        trans,
 #endif
 #elif defined(HYPRE_USING_DEVICE_OPENMP)
    hypre_CSRMatrixMatvecOMPOffload(trans, alpha, A, x, beta, y, offset);
+#elif defined(HYPRE_USING_HIP)
+#warning SPMV NOT IMPLEMENTED FOR HIP YET
```

Comment:
Is this ISO standard C?  If so, which standard?  If not, we should handle this differently.

Reply:
I do not believe it is officially standard, but it is supported by GCC, clang. I'm pretty sure it's supported by Intel as well. Regardless, this is a stop-gap.

On the one hand, I wanted the code in the PR to compile with the HIP compiler, but I did not want to mix in rocSPARSE support with this PR. On the other hand, HYPRE will not compile GPU support without a supported SpMV implementation. I thought this was a reasonable stop-gap to allow compilation, but still spit out something to alert the developer+user that they should not expect their program to run successfully on AMD GPUs yet.

I'm open to other suggestions.

Reply:
If this is just a stop-gap, we can leave it as is.  Thanks!

Reply:
We have `hypre_CSRMatrixSpMVDevice` implementation of SpMV. If SpMV is the only thing that stops us running on AMD GPUs, we can try it (we should add it anyway as it's marked as "TODO").

Reply:
> Is this ISO standard C? If so, which standard? If not, we should handle this differently.

I assume you two are talking about `#warning`. I didn't know it's not part of C. How about `#error`? Thanks.

Reply:
SpMV was only stopping the compiling. I believe SpGEMM is also required to actually run a test case? Either way, my own preference would be just get rocSPARSE in ASAP.

Reply:
> SpMV was only stopping the compiling. I believe SpGEMM is also required to actually run a test case? Either way, my own preference would be just get rocSPARSE in ASAP.

Yes, SpGEMM is needed for running AMG. We also have our own implementation for it (I am not sure if it has been compiled with HIP). I agree, getting rocSPARSE in hypre is probably easier to test first. Thanks!